ocean_provider/routes/consume.py

```diff
@@ -167,25 +167,26 @@ def initialize():
 
         token_address = service.datatoken_address
 
-        file_index = int(data.get("fileIndex"))
-        url_object = get_service_files_list(service, provider_wallet)[file_index]
-
-        url_valid, message = validate_url_object(url_object, service_id)
-
-        if not url_valid:
-            return (jsonify(error=message), 400)
-
-        download_url = get_download_url(url_object, app.config["PROVIDER_CONFIG_FILE"])
-        download_url = append_userdata(download_url, data)
-        valid, url_details = check_url_details(download_url)
-
-        if not valid:
-            logger.error(
-                f"Error: Asset URL not found or not available. \n"
-                f"Payload was: {data}",
-                exc_info=1,
+        file_index = int(data.get("fileIndex", "-1"))
```

Comment:
I would say just return an error message if file index is not ok, then un-indent the contents of the "if" statement

Reply:
this piece of code checks the validity of a file , as long as we have fileIndex defined.
If not, provider will return data for initialize, but without checking the file (maybe frontend already did that with fileinfo endpoint)